source/smart_pointer/smart_pointer.hpp

```diff
@@ -51,31 +67,69 @@ class SmartPointer {
    * SmartPointer. Si el conteo llega a cero, debe ser eliminado
    * (deallocated).
    */
-  SmartPointer &operator=(const SmartPointer &other) { return *this; }
-  SmartPointer &operator=(Type *other) { return *this; }
-  SmartPointer(const SmartPointer &other) {}
+  SmartPointer &operator=(const SmartPointer &other)
+  {
+    if (resource_ != other.resource_) // si el puntero es distinto tiene sentido
+    {
+      if (other.resource_ != nullptr) // creamos otro registro en tabla, o aumentamos el que tenemos
+        table_pointers[other.resource_]++;
+      if (resource_ != nullptr) // decrementamos nuestro registro actual
+        Detach();
+      resource_ = other.resource_;
+    }
+    return *this;
```

Comment:
Existe mucha indentación de los condicionales. Puedes simplificar la complejidad invirtiendo la logica

```suggestion
    if (resource_ == other.resource_) { // no se hace nada
        return *this;
    }
    if (other.resource_ != nullptr) { // creamos otro registro en tabla, o aumentamos el que tenemos
        table_pointers[other.resource_]++;
    }
    if (resource_ != nullptr) { // decrementamos nuestro registro actual
        Detach();
    }
    resource_ = other.resource_;
    return *this;
```